<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="pan_right_command"   default="90" />
    <arg name="pan_left_command"   default="0" />
    <arg name="pan_center_command"   default="45" />
    <arg name="pan_max_angle_deg"   default="45.0" />
    <arg name="pan_min_angle_deg"   default="-45.0" />

    <arg name="tilt_up_command"   default="0" />
    <arg name="tilt_down_command"   default="150" />
    <arg name="tilt_center_command"   default="100" />
    <arg name="tilt_max_angle_deg"   default="90.0" />
    <arg name="tilt_min_angle_deg"   default="-45.0" />

    <arg name="cmd_vel_topic" default="cmd_vel"/>
    <arg name="rover6_chassis_services_enabled" default="true"/>
    <arg name="rover6_chassis_use_sensor_msg_time" default="false"/>

    <group ns="rover6" >
        <node name="rover6_chassis" pkg="rover6_chassis" type="rover6_chassis_node.py" output="screen" required="true">
            <remap from="cmd_vel" to="$(arg cmd_vel_topic)" />
            <param name="services_enabled" value="$(arg rover6_chassis_services_enabled)"/>
            <param name="use_sensor_msg_time" value="$(arg rover6_chassis_use_sensor_msg_time)"/>

            <param name="pan_right_command"   value="$(arg pan_right_command)" />
            <param name="pan_left_command"   value="$(arg pan_left_command)" />
            <param name="pan_center_command"   value="$(arg pan_center_command)" />
            <param name="pan_max_angle_deg"   value="$(arg pan_max_angle_deg)" />
            <param name="pan_min_angle_deg"   value="$(arg pan_min_angle_deg)" />

            <param name="tilt_up_command"   value="$(arg tilt_up_command)" />
            <param name="tilt_down_command"   value="$(arg tilt_down_command)" />
            <param name="tilt_center_command"   value="$(arg tilt_center_command)" />
            <param name="tilt_max_angle_deg"   value="$(arg tilt_max_angle_deg)" />
            <param name="tilt_min_angle_deg"   value="$(arg tilt_min_angle_deg)" />
        </node>
    </group>
</launch>
